apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -0,0 +1,291 @@
+    const record: Record<string, ControlPreviewIssue[]> = {};
+
+    keys.forEach((key) => {
+      record[key] = [
+        {
+          issueType: ControlPreviewIssueTypeEnum.MISSING_VALUE,
+          message: `Value is missing on a required control`, // Custom message for the issue
+        },
+      ];
+    });
+
+    return record;
+  }
+  private findMissingKeys(requiredRecord: Record<string, unknown>, actualRecord: Record<string, unknown>) {
+    const requiredKeys = this.collectKeys(requiredRecord);
+    const actualKeys = this.collectKeys(actualRecord);
+
+    return difference(requiredKeys, actualKeys);
+  }
+  private collectKeys(obj, prefix = '') {
+    return reduce(
+      obj,
+      (result, value, key) => {
+        const newKey = prefix ? `${prefix}.${key}` : key;
+        if (isObject(value) && !isArray(value)) {
+          result.push(...this.collectKeys(value, newKey));
+        } else {
+          // Otherwise, just add the key
+          result.push(newKey);
+        }
+
+        return result;
+      },
+      []
+    );
+  }
+  private async executePreviewUsecase(
+    workflowId: string,
+    stepId: string,
+    origin: WorkflowOriginEnum,
+    hydratedPayload: Record<string, unknown>,
+    updatedControlValues: Record<string, unknown>,
+    command: GeneratePreviewCommand
+  ) {
+    return await this.legacyPreviewStepUseCase.execute(
+      PreviewStepCommand.create({
+        payload: hydratedPayload,
+        controls: updatedControlValues || {},
+        environmentId: command.user.environmentId,
+        organizationId: command.user.organizationId,
+        stepId,
+        userId: command.user._id,
+        workflowId,
+        workflowOrigin: origin,
+      })
+    );
+  }
+
+  private async getWorkflowUserIdentifierFromWorkflowObject(command: GeneratePreviewCommand) {
+    const workflowResponseDto = await this.getWorkflowUseCase.execute({
+      identifierOrInternalId: command.workflowId,
+      user: command.user,
+    });
+    const { workflowId, steps } = workflowResponseDto;
+    const step = steps.find((stepDto) => stepDto.stepUuid === command.stepUuid);
+    if (!step) {
+      throw new StepNotFoundException(command.stepUuid);
+    }
+
+    return {
+      workflowId,
+      stepId: step.slug,
+      stepType: step.type,
+      stepControlSchema: step.controls,
+      origin: workflowResponseDto.origin,
+    };
+  }
+
+  private payloadHydrationLogic(command: GeneratePreviewCommand) {
+    const dto = command.generatePreviewRequestDto;
+
+    let aggregatedDefaultValues = {};
+    const aggregatedDefaultValuesForControl: Record<string, Record<string, unknown>> = {};
+    const flattenedValues = flattenJson(dto.controlValues);
+    console.log('flattenedValues', flattenedValues);
+    for (const controlValueKey in flattenedValues) {
+      if (flattenedValues.hasOwnProperty(controlValueKey)) {
+        console.log('controlValueKey', controlValueKey);
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
```suggestion
```
☑ **chore:** Remove console.log‏